src/python/pants/engine/streaming_workunit_handler.py

```diff
@@ -180,54 +182,42 @@ def __init__(
         options_bootstrapper: OptionsBootstrapper,
         specs: Specs,
         report_interval_seconds: float,
+        pantsd: bool,
         max_workunit_verbosity: LogLevel = LogLevel.TRACE,
     ) -> None:
```

Comment:
So, it would still be possible to enter/exit this contextmanager multiple times, right? Using it in a `with` statement doesn't consume it.

One way to make that a non-issue though, maybe, would be to move the construction of the thread_runner into `__enter__`? That way you'd create a new one each time you called `__enter__`, and could fail if one already existed, etc.

Reply:
True, that's a good point that this doesn't actually prevent the problem from a language perspective.

However, my change does result in this error if you use the same handler twice:

```
15:00:25.87 [ERROR] threads can only be started once
Traceback (most recent call last):
  File "/Users/eric/code/pants/src/python/pants/bin/daemon_pants_runner.py", line 134, in single_daemonized_run
    return runner.run(start_time)
  File "/Users/eric/code/pants/src/python/pants/bin/local_pants_runner.py", line 255, in run
    with streaming_reporter:
  File "/Users/eric/code/pants/src/python/pants/engine/streaming_workunit_handler.py", line 213, in __enter__
    self.thread_runner.start()
  File "/Users/eric/.pyenv/versions/3.9.1/lib/python3.9/threading.py", line 869, in start
    raise RuntimeError("threads can only be started once")
RuntimeError: threads can only be started once
```